src/server-logger.ts

```diff
@@ -0,0 +1,257 @@
+import request from 'superagent';
+import stringify from 'safe-json-stringify';
+import cloneDeep from 'lodash.clonedeep';
+
+import { Logger } from './logger';
+import { IDetail, ILoggerConfig, LogLevel, ITrace, ILogBufferItem, ILogMessage } from './interfaces';
+import { calculateLogMessageSize } from './utils';
+
+const MAX_LOG_SIZE = 14500;
+const DEFAULT_UPLOAD_DEBOUNCE = 4000;
+
+export class ServerLogger {
```

Comment:
So we tried this approach with the last logger originally and found that having a ServerLogger class can cause some unintended consequences. For example in the case of volt, we'd have a two separate server loggers (one for volt and one for the sdk) both logging at the same time. We ran into issues with rate limiting. We then switched the server logger to be a singleton which was used by both apps and it would just batch send the logs. 

Having a configurable url, which I think is a good move, makes this a little more tricky. It almost seems like we need to have a ServerLoggerFactory that keeps track of the server loggers. if you request a server logger for a url (one that already exists) we just return that one, otherwise we can create a new one.

IIRC, the old logger used whatever debounce time was smallest between multiple apps. Just something to think about. I'm open to all ideas and arguments here.

Reply:
Updated PR. Now there can be individual `Logger`s (each will have its own `ServerLogger`). All `ServerLogger`s with the same `url` will use the same `LogUploader`.